### PR TITLE
Batch waiting state changes in engine

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this batches up waiting state changes so we only do them every half a second

### CODE REVIEW

File: shared/actions/json/waiting.json [hunk 1]
```diff
@@ -1,5 +1,4 @@
-{
-  "prelude": ["import type {RPCError} from '../util/errors'"],
+{ "prelude": ["import type {RPCError} from '../util/errors'"],
```
Comment (akalin-keybase) [inline shared/actions/json/waiting.json]: intended to do this?

File: shared/actions/json/waiting.json [hunk 2]
```diff
@@ -8,6 +7,9 @@
       "key": ["string", "Array<string>"],
       "error?": "RPCError"
     },
+    "batchChangeWaiting": {
+      "changes": "Array<{key: string | Array<string>, increment: boolean, error?: RPCError}>"
+    },
```
Comment (akalin-keybase) [inline shared/actions/json/waiting.json]: looks like changeWaiting is only used in one place (engine below) so we can get rid of it

Comment (akalin-keybase) [inline shared/actions/json/waiting.json]: also what about incrementWaiting / decrementWaiting? Do we care about batching those too?

Comment (chrisnojima) [reply shared/actions/json/waiting.json]: unclear. honestly most of these should hopefully go away by letting the engine do it. lemme add a pass of that cleanup in this pr

Comment (chrisnojima) [reply shared/actions/json/waiting.json]: good idea, removing

File: shared/actions/json/waiting.json (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/json/waiting.json]: nope

File: shared/actions/json/waiting.json (hunk 2 above)
Comment (chrisnojima) [reply shared/actions/json/waiting.json]: ok i remove a ton of manual increment in teams. we do explicitly increment / decrement but its only in 2 places really and not a perf matter

**Pull request merged**